File: bdm/src/main/webapp/js/model/reportpage.js
function ReportPage() {
    this.id = null;
    this.pageNo = 1;
    this.pageSize = 30;
    this.allEntitiesCount = 0;
    this.allPagesCount = 0;
    this.pageIndex = 0;
    this.pageEntities = [];
    this.titles = [];
    this.classtypename = "ReportPage";
};

ReportPage.prototype = new Page();

ReportPage.prototype.parseFromJSON = function(json) {
    this.id = json.id;
    this.pageNo = json.pageNo;
    this.pageSize = json.pageSize;
    this.allEntitiesCount = json.allEntitiesCount;
    this.allPagesCount = json.allPagesCount;
    this.pageIndex = json.pageIndex;
    if (json.titles.length > 0) {
        for (var i = 0; i < json.titles.length; i++) {
            var f = new ReportField();
            f.parsefromJSON(json.titles[i])
            this.titles.push(f);
        }
    }
    if (json.pageEntities.length > 0) {
        for (var i = 0; i < json.pageEntities.length; i++) {
            var row = new ReportDataRow();
            row.parsefromJSON(json.pageEntities[i]);
            this.pageEntities.push(row);
        }
    }
}

File: core.bigdata/src/main/java/com/xq/paas/core/bigdata/report/ReportPage.java
package com.xq.paas.core.bigdata.report;

import com.cloudibpm.core.Page;

/**
 * Pageable report model
 * @author Dahai Cao created at 14:26 on 2019-03-10
 */
public class ReportPage extends Page {
    private String id = null;// report service ID
    private String name = null;
    private ReportField [] titles = new ReportField[0];
    private ReportDataRow [] pageEntities = new ReportDataRow[0];
    private String classtypename = "ReportPage";

    public ReportPage() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getClasstypename() {
        return classtypename;
    }

    public void setClasstypename(String classtypename) {
        this.classtypename = classtypename;
    }

    public ReportPage(int pageNo, int pageSize) {
        super(pageNo, pageSize);
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public ReportField[] getTitles() {
        return titles;
    }

    public void setTitles(ReportField[] titles) {
        this.titles = titles;
    }

    public ReportDataRow[] getPageEntities() {
        return pageEntities;
    }

    public void setPageEntities(ReportDataRow[] pageEntities) {
        this.pageEntities = pageEntities;
    }
}
